test/libyul/yulOptimizerTests/fullSuite/aztec.yul

```diff
@@ -284,15 +284,17 @@
 //             let _5 := 0x40
 //             calldatacopy(0xe0, add(_3, 164), _5)
 //             calldatacopy(0x20, add(_3, 100), _5)
-//             mstore(0x120, sub(_2, c))
-//             mstore(0x60, k)
+//             let _6 := 0x120
+//             mstore(_6, sub(_2, c))
+//             let _7 := 0x60
+//             mstore(_7, k)
```

Comment:
That's weird - actually looks like *fewer* variables are eliminated.

Reply:
But ok - it might just be that choosing the "right" one first made it unnecessary to eliminate further ones in a second iteration... maybe wise to verify that that's what's happening let me check.

Reply:
Yes, as a matter of fact, this seems to be what's happening.
``develop`` used two iterations on the case, first eliminating 11 variables:
``_1982 _1986 _1989 _1993 _1949 _1943 _1955 _1932 _1930 _2020 _2048``
and then two more in a second iteration:
``_1990 _1925``

while after this PR, one iteration suffices, which eliminates these 11 instead, i.e. it chooses 
``_1982 _1986 _1993 _1949 _1943 _1955 _1925 _1932 _1930 _2020 _2048``
i.e. it doesn't choose _1989, but chooses _1925 in the first pass, which seems to solve the problem right away.

And it's probably not unlikely for this change to be beneficial like that - variables declared earlier have more space to be CSEd into, so maybe more likely to be used more often and put more strain on the stack.

Or in summary: seems fine ;-).

Reply:
By the way I don't think the old version aborted here due to a variable being referenced, but the stack error just wasn't fixed by the choice it made (there's no guarantee that it works in the old code transform - it's more like guessing what might help and then running again to see if there's still issues :-) - the new code transform will be able to tell more accurately what helps and what doesn't)